packages/react-instantsearch-core/src/components/__tests__/InstantSearch.js

```diff
@@ -7,17 +7,31 @@ import { InstantSearchConsumer } from '../../core/context';
 
 Enzyme.configure({ adapter: new Adapter() });
 
+const createFakeInstantSearchManager = (rest = {}) => ({
+  context: {},
+  updateIndex: jest.fn(() => {}),
+  updateClient: jest.fn(() => {}),
+  ...rest,
+});
+
+const createFakeSearchClient = (rest = {}) => ({
+  search: jest.fn(() => {}),
+  ...rest,
+});
+
 jest.mock('../../core/createInstantSearchManager', () =>
   jest.fn(() => ({
     context: {},
+    updateIndex: () => {},
+    updateClient: () => {},
```

Comment:
It would be much simpler to test with the manager as prop. It might be an improvement for a latter PR.